### PR TITLE
Add Consumer reconciler tests

### PR DESCRIPTION
**Pull request by pierDipi**
Part of #1537 

## Proposed Changes

- Add Consumer reconciler tests

TODO:
- finalizer tests
- more reconciler tests

### CODE REVIEW

File: control-plane/pkg/reconciler/consumer/consumer_test.go [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+			Objects: []runtime.Object{
+				NewService(),
+				NewConsumerGroup(ConsumerGroupOwnerRef(SourceAsOwnerReference())),
+				NewDispatcherPod("p1", PodRunning()),
+				NewConsumer(1,
+					ConsumerUID(ConsumerUUID),
+					ConsumerSpec(NewConsumerSpec(
+						ConsumerTopics("t1", "t2"),
+						ConsumerConfigs(
+							ConsumerBootstrapServersConfig(ChannelBootstrapServers),
```
Comment (aavarghese) [inline control-plane/pkg/reconciler/consumer/consumer_test.go:71]: Can we change this to `SourceBootstrapServers`?

File: control-plane/pkg/reconciler/consumer/consumer_test.go [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+					DataPlaneConfigFormat:       base.Json,
+					DataPlaneConfigMapNamespace: SystemNamespace,
+					DataPlaneConfigMapName:      "p1",
+				}, &contract.Contract{
+					Generation: 1,
+					Resources: []*contract.Resource{
+						{
+							Uid:              ConsumerUUID,
+							Topics:           []string{"t1", "t2"},
+							BootstrapServers: ChannelBootstrapServers,
```
Comment (aavarghese) [inline control-plane/pkg/reconciler/consumer/consumer_test.go:107]: Can we change this to `SourceBootstrapServers`?

File: control-plane/pkg/reconciler/consumer/consumer_test.go [hunk 3]
```diff
@@ -0,0 +1,379 @@
...
+			},
+			WantStatusUpdates: []clientgotesting.UpdateActionImpl{
+				{
+					Object: func() runtime.Object {
+						c := NewConsumer(1,
+							ConsumerUID(ConsumerUUID),
+							ConsumerSpec(NewConsumerSpec(
+								ConsumerTopics("t1", "t2"),
+								ConsumerConfigs(
+									ConsumerBootstrapServersConfig(ChannelBootstrapServers),
```
Comment (aavarghese) [inline control-plane/pkg/reconciler/consumer/consumer_test.go:148]: Can we change this to `SourceBootstrapServers`?

File: control-plane/pkg/reconciler/consumer/consumer_test.go [hunk 4]
```diff
@@ -0,0 +1,379 @@
...
+			Objects: []runtime.Object{
+				NewService(),
+				NewConsumerGroup(ConsumerGroupOwnerRef(SourceAsOwnerReference())),
+				NewDispatcherPod("p1", PodRunning()),
+				NewConsumer(1,
+					ConsumerUID(ConsumerUUID),
+					ConsumerSpec(NewConsumerSpec(
+						ConsumerTopics("t1", "t2"),
+						ConsumerConfigs(
+							ConsumerBootstrapServersConfig(ChannelBootstrapServers),
```
Comment (aavarghese) [inline control-plane/pkg/reconciler/consumer/consumer_test.go:177]: Can we change this to `SourceBootstrapServers`?

File: control-plane/pkg/reconciler/consumer/consumer_test.go [hunk 5]
```diff
@@ -0,0 +1,379 @@
...
+					DataPlaneConfigFormat:       base.Json,
+					DataPlaneConfigMapNamespace: SystemNamespace,
+					DataPlaneConfigMapName:      "p1",
+				}, &contract.Contract{
+					Generation: 1,
+					Resources: []*contract.Resource{
+						{
+							Uid:              ConsumerUUID,
+							Topics:           []string{"t1", "t2"},
+							BootstrapServers: ChannelBootstrapServers,
```
Comment (aavarghese) [inline control-plane/pkg/reconciler/consumer/consumer_test.go:220]: Can we change this to `SourceBootstrapServers`?

File: control-plane/pkg/reconciler/consumer/consumer_test.go [hunk 6]
```diff
@@ -0,0 +1,379 @@
...
+			},
+			WantStatusUpdates: []clientgotesting.UpdateActionImpl{
+				{
+					Object: func() runtime.Object {
+						c := NewConsumer(1,
+							ConsumerUID(ConsumerUUID),
+							ConsumerSpec(NewConsumerSpec(
+								ConsumerTopics("t1", "t2"),
+								ConsumerConfigs(
+									ConsumerBootstrapServersConfig(ChannelBootstrapServers),
```
Comment (aavarghese) [inline control-plane/pkg/reconciler/consumer/consumer_test.go:267]: Can we change this to `SourceBootstrapServers`?

File: control-plane/pkg/reconciler/consumer/consumer_test.go [hunk 7]
```diff
@@ -0,0 +1,379 @@
...
+			Name: "Pod not found",
+			Objects: []runtime.Object{
+				NewService(),
+				NewConsumerGroup(ConsumerGroupOwnerRef(SourceAsOwnerReference())),
+				NewConsumer(1,
+					ConsumerUID(ConsumerUUID),
+					ConsumerSpec(NewConsumerSpec(
+						ConsumerTopics("t1", "t2"),
+						ConsumerConfigs(
+							ConsumerBootstrapServersConfig(ChannelBootstrapServers),
```
Comment (aavarghese) [inline control-plane/pkg/reconciler/consumer/consumer_test.go:303]: Can we change this to `SourceBootstrapServers`?

File: control-plane/pkg/reconciler/consumer/consumer_test.go [hunk 8]
```diff
@@ -0,0 +1,379 @@
...
+			},
+			WantStatusUpdates: []clientgotesting.UpdateActionImpl{
+				{
+					Object: func() runtime.Object {
+						c := NewConsumer(1,
+							ConsumerUID(ConsumerUUID),
+							ConsumerSpec(NewConsumerSpec(
+								ConsumerTopics("t1", "t2"),
+								ConsumerConfigs(
+									ConsumerBootstrapServersConfig(ChannelBootstrapServers),
```
Comment (aavarghese) [inline control-plane/pkg/reconciler/consumer/consumer_test.go:329]: Can we change this to `SourceBootstrapServers`?

File: control-plane/pkg/reconciler/consumer/consumer_test.go [hunk 9]
```diff
@@ -0,0 +1,379 @@
...
+				NewService(),
+				NewConsumerGroup(ConsumerGroupOwnerRef(SourceAsOwnerReference())),
+				NewDispatcherPod("p1", PodRunning()),
+				NewConsumer(1,
+					ConsumerUID(ConsumerUUID),
+					ConsumerSpec(NewConsumerSpec(
+						ConsumerTopics("t1", "t2"),
+						ConsumerConfigs(
+							ConsumerBootstrapServersConfig(ChannelBootstrapServers),
+							ConsumerGroupIdConfig("my.group.id"),
```
Comment (aavarghese) [inline control-plane/pkg/reconciler/consumer/consumer_test.go:72]: And all of these to `SourceConsumerGroup` for `ConsumerGroupIdConfig` fields?

File: control-plane/pkg/reconciler/consumer/consumer_test.go [hunk 10]
```diff
@@ -0,0 +1,379 @@
...
+		{
+			Name: "Reconciled normal",
+			Objects: []runtime.Object{
+				NewService(),
+				NewConsumerGroup(ConsumerGroupOwnerRef(SourceAsOwnerReference())),
+				NewDispatcherPod("p1", PodRunning()),
+				NewConsumer(1,
+					ConsumerUID(ConsumerUUID),
+					ConsumerSpec(NewConsumerSpec(
+						ConsumerTopics("t1", "t2"),
```
Comment (aavarghese) [inline control-plane/pkg/reconciler/consumer/consumer_test.go:69]: `SourceTopics` if you'd like..

**Review by aavarghese — COMMENTED** [review]
minor nits since Source is the Owner for the CGs in these tests..

**Comment by pierDipi** [comment]
Thanks @aavarghese !

**Review by aavarghese — APPROVED** [review]
/lgtm

**Comment by devguyio** [comment]
Looks good! Need to rebase

/lgtm

**Comment by matzew** [comment]
needs a rebase, @pierDipi

**Comment by pierDipi** [comment]
Rebased!

**Review by aavarghese — APPROVED** [review]
/lgtm

**Pull request merged**